torch/_inductor/codegen/cpp_wrapper_cpu.py

```diff
@@ -2204,6 +2204,17 @@ def generate_py_arg_inner(raw_arg, arg_type):
                 return f"PyBool_FromLong({1 if raw_arg else 0})"
             elif isinstance(arg_type, torch.StringType):
                 return f'PyUnicode_FromString("{raw_arg}")'
+            elif isinstance(arg_type, torch.NumberType):
+                # Union[bool, int, float, complex]
+                # torch/_prims_common/__init__.py
+                if isinstance(raw_arg, int):
+                    return f"PyLong_FromLongLong({raw_arg})"
+                elif isinstance(raw_arg, float):
+                    return f"PyFloat_FromDouble({raw_arg})"
+                elif isinstance(raw_arg, bool):
+                    return f"PyLong_FromLong({raw_arg})"
```

Comment:
You can use the  `return f"PyBool_FromLong({1 if raw_arg else 0})"` from the earlier`elif isinstance(arg_type, torch.BoolType):` branch.